VL.Data/Services/AccountService.cs

```diff
@@ -0,0 +1,104 @@
+﻿using Microsoft.AspNetCore.DataProtection;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Options;
+using Microsoft.IdentityModel.Tokens;
+using System.IdentityModel.Tokens.Jwt;
+using System.Runtime.InteropServices;
+using System.Security.Claims;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using VL.Shared.Data;
+using VL.Shared.Interfaces;
+using VL.Shared.Model;
+
+namespace VL.Shared.Services
+{
+    public class AccountService : IAccountService
+    {
+        private readonly UserManager<IdentityUser> _userManager;
+        private readonly SignInManager<IdentityUser> _signInManager;
+        private readonly RoleManager<IdentityRole> _roleManager;
+        private readonly IConfiguration _configuration;
+        private readonly ApplicationDbContext _applicationDbContext;
+
+        public AccountService(UserManager<IdentityUser> userManager,
+            SignInManager<IdentityUser> signInManager,
+            RoleManager<IdentityRole> roleManager,
+            IConfiguration configuration,
+            ApplicationDbContext applicationDbContext)
+        {
+            _userManager = userManager;
+            _signInManager = signInManager;
+            _roleManager = roleManager;
+            _configuration = configuration;
+            _applicationDbContext = applicationDbContext;
+        }
+
+        public async Task<IdentityResult> Register(UserDto userDto)
+        {
+            var user = new IdentityUser
+            {
+                UserName = userDto.Email,
+                Email = userDto.Email
+            };
+
+            var userResult = await _userManager.CreateAsync(user, userDto.Password);
+
+            if (userResult.Succeeded)
+            {
+                await _userManager.AddToRoleAsync(user, userDto.Role);
+            }
+            return userResult;
+        }
+
+        public async Task<string> Login(LoginDto userDto)
```

Comment:
If you are not going to return a specific error string as a result of your signInManager attempt this might as well return a boolean

Some flows you could return the actual error, 
"Email or username does not exist"
"Invalid password"
etc.

This opens up a little more insight to the user but is also useful information to a bad actor depending on the application (is it open facing, is it an internal company client, etc.)

Reply:
I chose to return a string in order to pass my Jwt. If I change the tag on the task to return a bool, I am not sure how I would pass the Jwt forward.

Reply:
You may want to abstract that as you are now also trying to generate and return a token for when a user registers for example